Bump JAX libs for CI

mihaimaruseac opened the pull request:
JAX libs are needed in CI as part of testing JAX <-> TFLite integration. This is only needed on Linux and Mac, as the corresponding tests are not run on Windows at this time.

There is no need for an equivalent change in `setup.py` as TF wheel does not depend on JAX. This is just a test-only dependency.

Fixes #56699

mihaimaruseac committed:
Bump JAX libs for CI

JAX libs are needed in CI as part of testing JAX <-> TFLite integration. This is only needed on Linux and Mac, as the corresponding tests are not run on Windows at this time.

There is no need for an equivalent change in `setup.py` as TF wheel does not depend on JAX. This is just a test-only dependency

kulinseth reviewed (approved):
Looks fine to me. Should we take @hawkinsp 's advice and make it:
`jax ~= 0.3.14`

mihaimaruseac reviewed (commented):
You are right, @kulinseth . will send a fix commit.

(apparently, I need to learn how to use the web ui more to do a commit instead of posting a comment)

mihaimaruseac committed:
Pin to minor version.

Without the pin to minor version, when `jax`/`jaxlib` would release version 0.4 CI would pick it up, but that might contain breaking changes. With the pin to the minor version and mentioning the explicit patch number 14, only 0.3.14, 0.3.15, ... versions would be allowed, so a better compatibility story
